tests/components/broadlink/test_remote.py

```diff
@@ -0,0 +1,138 @@
+"""Tests for Broadlink remotes."""
+from base64 import b64decode
+
+from homeassistant.components.broadlink.const import DOMAIN, REMOTE_DOMAIN
+from homeassistant.components.remote import (
+    SERVICE_SEND_COMMAND,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.entity_registry import async_entries_for_device
+
+from . import get_device
+
+from tests.async_mock import call, patch
+from tests.common import mock_device_registry, mock_registry
+
+REMOTE_DEVICES = ["Entrance", "Living Room", "Office", "Garage"]
+
+IR_PACKET = (
+    "JgBGAJKVETkRORA6ERQRFBEUERQRFBE5ETkQOhAVEBUQFREUEBUQ"
+    "OhEUERQRORE5EBURFBA6EBUQOhE5EBUQFRA6EDoRFBEADQUAAA=="
+)
+
+
+async def test_remote_setup_works(hass):
```

Comment:
```suggestion
def _patch_broadlink_gendevice(return_value):
    """Patch the broadlink gendevice method."""
    return patch(
    .   "homeassistant.components.broadlink.device.blk.gendevice", return_value=return_value
    )


async def test_remote_setup_works(hass):
```

Reply:
I think we should keep return_value as a keyword argument because sometimes we may want to use a side_effect instead, so we always keep the behavior explicit at the same level.

Reply:
fair point, updated suggestions

Reply:
I think the current syntax is cleaner and more flexible. It helps us to grow this code faster by allowing us to easily copy, paste and change the parts that we want in order to create more tests for other parts of the code.

Sometimes we may want to patch broadlink.discover instead of broadlink.gendevice. So we just copy, paste and change "gendevice" to "discover". The proposed change bricks the code by turning the argument "broadlink.gendevice" into a hardcoded function. So if we wanted to patch broadlink.discover instead of broadlink.gendevice we would need to create an auxiliary function for that as well, an so forth. We would end up with a lot of auxiliary functions that do nothing but hiding arguments that should be visible in the first place.

I think that all patches should be explicit at the same level in a clear, direct and universal way so the reader can see what is going on even if he does not know this codebase.

Reply:
the abstraction is optional though helps with reviewing since you then dont have a bunch of patch targets to verify. You could also abstract the setup of config entry so you only patch gendevice in one place, again optional.

you will need to use "homeassistant.components.broadlink.device.blk.gendevice" though because patching the library leads to potential issues in tests. https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
Oh, sure, your suggestions make a lot more sense to me now. I didn't know I couldn't patch the library directly, thanks for informing me about this. I'll look at the code later and see how I can improve it.

Reply:
What did you think? Better huh? Do you allow me to fix the other tests in this PR?

Reply:
we should do others in new PR to keep things focused

Reply:
Ok.